forge-game/src/main/java/forge/game/ability/effects/ClashEffect.java

```diff
@@ -33,7 +33,10 @@ protected String getStackDescription(final SpellAbility sa) {
     public void resolve(final SpellAbility sa) {
         final Card source = sa.getHostCard();
         final Player player = source.getController();
-        final Player opponent = sa.getActivatingPlayer().getController().chooseSingleEntityForEffect(player.getOpponents(), sa, Localizer.getInstance().getMessage("lblChooseOpponent"), null);
+        final Player opponent = sa.hasParam("Defined") ?
+                AbilityUtils.getDefinedPlayers(source, sa.getParam("Defined"), sa).getFirst() :
```

Comment:
@tool4ever you plan to make this cleaner? or is this okay?

Reply:
safe for now I think